internal/controller/pdb_to_pdbwatcher_controller.go

```diff
@@ -0,0 +1,179 @@
+			return reconcile.Result{}, e
+		}
+
+		// Create a new PDBWatcher
+		pdbWatcher = types.PDBWatcher{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "PDBWatcher",
+				APIVersion: "apps.mydomain.com/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pdb.Name,
+				Namespace: pdb.Namespace,
+				Annotations: map[string]string{
+					"controlledBy": "PDBToPDBWatcherController",
+					"target":       deploymentName,
+				},
+			},
+			Spec: types.PDBWatcherSpec{
+				TargetName: deploymentName,
+				TargetKind: deploymentKind,
+			},
+		}
+
+		err := r.Create(ctx, &pdbWatcher)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("unable to create PDBWatcher: %v", err)
+		}
+		log.Log.Info("Created PDBWatcher", "name", pdb.Name)
+	}
+
+	// Return no error and no requeue
+	return reconcile.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PDBToPDBWatcherReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Set up the controller to watch Deployments and trigger the reconcile function
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&policyv1.PodDisruptionBudget{}).
+		WithEventFilter(predicate.Funcs{
+			// Only trigger for Create and Delete events
+			CreateFunc: func(e event.CreateEvent) bool {
+				// Handle create event (this will be true for all create events)
+				//should we create pdbwatchers for customer created pdbs
+				return true
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				// Handle delete event (this will be true for all delete events)
+				return true
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				//ToDo: theoretically you could have a pdb update and change
+				// its label selectors in which case you might need to update the deployment target?
+				return false
+			},
+		}).
+		Owns(&types.PDBWatcher{}). // Watch PDBs for ownership
+		Complete(r)
+}
+
+func (r *PDBToPDBWatcherReconciler) discoverDeployment(ctx context.Context, pdb *policyv1.PodDisruptionBudget) (string, error) {
+	logger := log.FromContext(ctx)
+
+	// Convert PDB label selector to Kubernetes selector
+	selector, err := metav1.LabelSelectorAsSelector(pdb.Spec.Selector)
+	if err != nil {
+		return "", fmt.Errorf("error converting label selector: %v", err)
+	}
+	logger.Info("PDB Selector", "selector", pdb.Spec.Selector)
+
+	podList := &corev1.PodList{}
+	err = r.List(ctx, podList, &client.ListOptions{Namespace: pdb.Namespace, LabelSelector: selector})
+	if err != nil {
+		return "", fmt.Errorf("error listing pods: %v", err)
+	}
+	logger.Info("Number of pods found", "count", len(podList.Items))
+
+	if len(podList.Items) == 0 {
+		return "", fmt.Errorf("no pods found matching the PDB selector %s", pdb.Name)
+	}
+
+	// Iterate through each pod
+	for _, pod := range podList.Items {
+		logger.Info("Examining pod", "pod", pod.Name)
+
+		// Check the OwnerReferences of each pod
+		for _, ownerRef := range pod.OwnerReferences {
+			if ownerRef.Kind == "ReplicaSet" {
```

Comment:
remind me to make consts for these later. I also hardcoded strings somewhere I shouldn't have.